### PR TITLE
Bens backend connection

### PR DESCRIPTION
**Pull request by BenjaminWalkerBond**
Rudimentary connection between front end and backend without data persistence on frontend(yet).

### CODE REVIEW

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/controller/distribution.controller.js]:
```suggestion
    //Save to postgresql database
```

Comment spelling 👨‍🎨

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/controller/distribution.controller.js]: You should be able to use async/await here to be more readable than calling Promise methods directly:

https://javascript.info/async-await

So this handler would become:

```js
exports.create = async (req, res) => {
  try {
    const distribution = await Distribution.create({
      deliveryLocation: req.body.deliveryLocation,
      deliveryDepartureTime: req.body.deliveryWeights,
      projectedNumberOfClients: req.body.projectedNumberOfClients,
    });
    res.json(distribution);
  } catch (err) {
    console.log(err);
    res.status(500).json({msg: "error", details: err});
  }
}
```

Right now it's pretty readable either way, but async/await will help keep things that way as we increase complexity.

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/controller/distribution.controller.js]: We should also be able to extract this common `catch()` clause ☝️  to a reusable function. (Maybe in a follow-on PR.)

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/controller/distribution.controller.js]: It looks like this endpoint will return an array; empty if `deliveryLocation` is not found, containing one element if it is. It would be a little more idiomatic RESTful to return a single result or a 404 instead.

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/controller/distribution.controller.js]:
```suggestion
			res.json(distributions.sort((c1, c2) => c1.id - c2.id));
```

Just a tad terser. (And a little less surprising if you end up needing to use `this` in here for some reason later.)

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/controller/distribution.controller.js]: How does this fail is `deliveryDepartureTime` is absent, or malformed?

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/controller/distribution.controller.js]:
```suggestion
				res.status(200).json( { msg: "Updated Successfully"} );
```

Typo :smile:

File: hcfb-be/backend-server/app/model/distribution.model.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/model/distribution.model.js]:
```suggestion
                    if(value.length !== 7) {
```

Triple-unequals to avoid the type coercion here.

File: hcfb-be/backend-server/app/model/rescue.model.js
Comment (smashwilson) [inline hcfb-be/backend-server/app/model/rescue.model.js]: 🍖

File: hcfb-fe/src/app/rescue/rescue-add.component.ts
Comment (smashwilson) [inline hcfb-fe/src/app/rescue/rescue-add.component.ts]: "getLocation()" sounds like a method that's intended to return something, while this initializes the subscription instead. How about something like `subscribeToLocation` instead?

File: hcfb-fe/src/app/rescue/rescue.service.ts
Comment (smashwilson) [inline hcfb-fe/src/app/rescue/rescue.service.ts]: It'd be nice to be able to have an actual type here 🤔 Ideally, one that's guaranteed to stay up to date with the actual type returned by the backend API.

I think being able to do that well would take a lot of work, though - likely extracting the model to an independent module and porting it to TypeScript.

File: hcfb-fe/src/app/rescue/rescue.service.ts
Comment (smashwilson) [inline hcfb-fe/src/app/rescue/rescue.service.ts]: Recommendation: let's extract URL construction into its own object. Then we can tackle pointing this at the real backend in production, and making it configurable for local development, without having to find-replace all over the place.

**Review by smashwilson — APPROVED** [review]
A few suggestions and tweaks here or there but overall a good looking start :+1:

Especially since we aren't continuously deploying anywhere yet, I say merge aggressively and improve things with more PRs. 😁

**Comment by smashwilson** [comment]
A sidenote: I tend to be pretty chatty in code reviews. Otherwise reviews can easily devolve into "here's a list of stuff you screwed up!" which can feel pretty dispiriting! A side effect is that you shouldn't feel like every comment is "must fix before merge"... if there _is_ something that I feel we shouldn't merge without fixing, I'll leave a "request changes" review and highlight those with 🚨or something to make it clear.

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (BenjaminWalkerBond) [reply hcfb-be/backend-server/app/controller/distribution.controller.js]: That makes plenty of sense! Does this mean that I should refactor all controller code to utilize async and await keywords?

**Comment by BenjaminWalkerBond** [comment]
Hey Ash! I was wondering how I should go about changing all this code? Should I make all the changes and then submit another pull request? Am I able to make the changes in this pull request?

**Comment by smashwilson** [comment]
> Am I able to make the changes in this pull request?

Oh yeah. You can add changes to this pull request by making further commits on this branch (`bens_backend_connection`) and pushing them up with `git push`.

File: hcfb-be/backend-server/app/controller/distribution.controller.js
Comment (smashwilson) [reply hcfb-be/backend-server/app/controller/distribution.controller.js]: "should" is a strong word, haha. If you have the time and energy for it... it would also be totally fine to merge this as-is and refactor them in later PRs.

**Review by bluek1te — APPROVED** [review]

**Comment by bluek1te** [comment]
You can merge, we can make separate issues to remember to change the requests to async.

**Comment by bluek1te** [comment]
Normally you would want to merge into develop though for next time.

**Pull request merged**
